### ISSUE TITLE
Delete a file in a bucket

### ISSUE THREAD

**Issue by pijpijpij** [spring-cloud/spring-cloud-gcp#2407]
The users of my web-app have the ability to upload a few files. Our app needs to process them before uploading then a GCP bucket, in a small folder hierarchy. Today the Spring Resource API of `GoogleStorageResource` allows it to create them.
Our users also need to be able delete those files when they are not longer used. 

Giving my users direct access to GCP is of course not a viable option.

I cannot find an API in `GoogleStorageResource`  that would allow deleting them or the folders they are stored in. When our resource storage facility is backed with the local file system instead of GCP, I can use `getFile()` to get to `File.delete()`.
Would it be possible to add a deletion API to `GoogleStorageResource`  so I can delete those files and folders?

**Comment by meltsufin** [comment]
Have you tried using the `Blob`?
```java
GoogleStorageResource file = ...;
file.getBlob().delete();
```

**Comment by pijpijpij** [comment]
I did miss that, so thank you.
Unfortunately when I tried it `getBlob()` returned `null` for folders, so I also learned that folders are not first class citizens.
Instead I now iterate over all the blobs in the bucket of my folder and delete those whose `blobName` starts with my folder's `blobName`. When all the blobs in the folder are deleted, the folder magically dissappears!
Thanks for the pointer, it did take me out of my dead-end.
